client/src/redux/donation-saga.js

```diff
@@ -45,27 +46,29 @@ const updateCardErrorMessage = i18next.t('donate.error-3');
 
 function* showDonateModalSaga() {
   let shouldRequestDonation = yield select(shouldRequestDonationSelector);
-  const recentlyClaimedBlock = yield select(recentlyClaimedBlockSelector);
   const MODAL_SHOWN_KEY = 'modalShownTimestamp';
   const modalShownTimestamp = sessionStorage.getItem(MODAL_SHOWN_KEY);
   const isModalRecentlyShown = Date.now() - modalShownTimestamp < 20000;
-  if (
-    shouldRequestDonation &&
-    recentlyClaimedBlock &&
-    recentlyClaimedBlock.superBlock === 'full-stack-developer'
-  ) {
-    yield put(preventBlockDonationRequests());
-  } else if (shouldRequestDonation || isModalRecentlyShown) {
+  const shouldShowModal = shouldRequestDonation || isModalRecentlyShown;
+  const donatableSectionRecentlyCompleted = yield select(
+    donatableSectionRecentlyCompletedSelector
+  );
+
+  if (shouldShowModal) {
     yield delay(200);
     yield put(openDonationModal());
     sessionStorage.setItem(MODAL_SHOWN_KEY, Date.now());
     yield take(appTypes.closeDonationModal);
-    if (recentlyClaimedBlock) {
-      yield put(preventBlockDonationRequests());
-    } else {
+    if (!donatableSectionRecentlyCompleted) {
       yield call(saveCurrentCount);
     }
   }
+
+  /* users can complete donatable section but have less than 10 completed challenge
+     to show the donation modal.*/
+  if (donatableSectionRecentlyCompleted) {
+    yield put(preventSectionDonationRequests());
+  }
```

Comment:
What is the intended behaviour?

Do we want to show donations

* immediately after completing a section if they've also completed > 10 challenges.
* immediately after completing a challenge if they've also recently completed a section.

Reply:
the first one.